src/properties/propertiesService.ts

```diff
@@ -27,6 +29,38 @@ interface PropertySchemas {
   uiSchema?: UISchemaElement;
 }
 
+const ruleSchema = {
+  rule: {
+    type: 'object',
+  },
+};
+
+const ruleUiSchema = {
+  type: 'Control',
+  scope: '#/properties/rule',
+};
+
+const withRule = (schemas: PropertySchemas) => {
+  const schema = schemas.schema.properties
+    ? schemas.schema
+    : {
+        type: 'object',
+        properties: {},
+      };
+  const uiSchema =
+    schemas.uiSchema && isLayout(schemas.uiSchema)
+      ? schemas.uiSchema
+      : {
+          type: 'VerticalLayout',
+          elements: [],
+        };
```

Comment:
The checks and initializiations in `withRule` look weird to me as we have the `schemas` completely in our control. They also lead to a bug: The text of labels can no longer be changed in the properties view.

I would like to suggest to structure this a little bit differently:

In `getProperties` create the default empty schema object and empty vertical layout schemas, then call `withRule` on them. Now you know that the schema and ui schema always exist and you only need to slighly modify them. 

Similarity you can do `withText` for the label, `withLabel` for the group etc.

Reply:
For #68 I changed the PropertiesService to align more with the changes we planned for #53. I pushed those changes here, to make things easier. I will adress the rest of the review comments in a separate commit.